### PR TITLE
External keyboard bug Closes #1648

### LINKED ISSUE #1648: When hardware (bluetooth) keyboard is connected, collection view contentInset.bottom is wrong
1. Connect apple bluetooth keyboard on iPad
2. send messages
3. Note that `contentInset.bottom` and `scrollIndicatorInsets.bottom` are incorrect (too large)

### PR DESCRIPTION
**Pull request by MacMeDan**
## Pull request checklist

- [X] All tests pass. Demo project builds and runs.
- [X] I have resolved any merge conflicts.
- [X] I have followed the [coding style](https://github.com/jessesquires/HowToContribute#style-guidelines), and reviewed the [contributing guidelines](https://github.com/jessesquires/JSQMessagesViewController/blob/develop/.github/CONTRIBUTING.md). Confirmation: 💪😎👊

## What's in this pull request?

> I reworked the animation of the keyboard. Now it animates up if the `QuickTextBar` shows up or is dismissed and handles bluetooth connected keyboard. also scrolls up as you receive and send new messages.

### CODE REVIEW

**Comment by MacMeDan** [comment]
@jessesquires is the codecov/patch bot checking test coverage?

File: JSQMessagesViewController/Controllers/JSQMessagesViewController.m [hunk 1]
```diff
@@ -914,27 +914,25 @@ - (void)jsq_registerForNotifications:(BOOL)registerForNotifications
...
-    double animationDuration = [userInfo[UIKeyboardAnimationDurationUserInfoKey] doubleValue];
-
-    [UIView animateWithDuration:animationDuration
-                          delay:0.0
-                        options:animationCurveOption
-                     animations:^{
-                         [self jsq_setCollectionViewInsetsTopValue:self.collectionView.contentInset.top
-                                                       bottomValue:CGRectGetHeight(keyboardEndFrame)];
-                     }
-                     completion:nil];
```
Comment (jessesquires) [inline JSQMessagesViewController/Controllers/JSQMessagesViewController.m]: we need to keep all of the above 😄

File: JSQMessagesViewController/Controllers/JSQMessagesViewController.m [hunk 2]
```diff
@@ -914,27 +914,25 @@ - (void)jsq_registerForNotifications:(BOOL)registerForNotifications
...
+    CGRect keyboardBeginFrame = [[[notification userInfo] objectForKey:UIKeyboardFrameBeginUserInfoKey] CGRectValue];
+    CGRect keyboardEndFrame = [[[notification userInfo] objectForKey:UIKeyboardFrameEndUserInfoKey] CGRectValue];
+    
+    UIViewAnimationCurve animationCurve = [[[notification userInfo] objectForKey:UIKeyboardAnimationCurveUserInfoKey] integerValue];
+    NSTimeInterval animationDuration = [[[notification userInfo] objectForKey:UIKeyboardAnimationDurationUserInfoKey] integerValue];
+    
+    [UIView beginAnimations:nil context:nil];
+    [UIView setAnimationDuration:animationDuration];
+    [UIView setAnimationCurve:animationCurve];
```
Comment (jessesquires) [inline JSQMessagesViewController/Controllers/JSQMessagesViewController.m]: this is old school UIView animations. the block-based methods (above) have been preferred for awhile.

File: JSQMessagesViewController/Controllers/JSQMessagesViewController.m [hunk 3]
```diff
@@ -914,27 +914,25 @@ - (void)jsq_registerForNotifications:(BOOL)registerForNotifications
...
+    
+    CGRect newFrame = self.view.frame;
+    CGRect keyboardFrameBegin = [self.view convertRect:keyboardBeginFrame toView:nil];
+    CGRect keyboardFrameEnd = [self.view convertRect:keyboardEndFrame toView:nil];
+    
+    
+    newFrame.origin.y -= (keyboardFrameBegin.origin.y - keyboardFrameEnd.origin.y);
+    self.view.frame = newFrame;
```
Comment (jessesquires) [inline JSQMessagesViewController/Controllers/JSQMessagesViewController.m]: we don't want to adjust view frames at all.

It's  `contentInset.bottom` and `scrollIndicatorInsets.bottom` that are incorrect

**Comment by jessesquires** [comment]
hey @MacMeDan - let's close this and re-open with a different fix, this isn't going to work 😅

**Comment by jessesquires** [comment]
i think all we need to do is compute the keyboard frame diff (like you have) and pass that value to 

`jsq_setCollectionViewInsetsTopValue: bottomValue:`

**Comment by MacMeDan** [comment]
For some reason I did not even see these comments till now. Ya I will take another look at it.

**Pull request closed without merging**